source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -1245,5 +1245,155 @@ public void TestParsingRichTextBlockWithInvalidInlineType()
+                            ""wrap"": true
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            ""type"": ""TextBlock"",
+            ""text"": ""Now that we have defined the main rules and features of the format, we need to produce a schema and publish it to GitHub. The schema will be the starting point of our reference documentation."",
+            ""wrap"": true
+        },
+        {
+            ""type"": ""FactSet"",
+            ""facts"": [
+                {
+                    ""title"": ""Board:"",
+                    ""value"": ""Adaptive Card""
+                },
+                {
+                    ""title"": ""List:"",
+                    ""value"": ""Backlog""
+                },
+                {
+                    ""title"": ""Assigned to:"",
+                    ""value"": ""Matt Hidinger""
+                },
+                {
+                    ""title"": ""Due date:"",
+                    ""value"": ""Not set""
+                }
+            ]
+        }
+    ],
+    ""actions"": [
+        {
+            ""type"": ""Action.ShowCard"",
+            ""title"": ""Set due date"",
+            ""card"": {
+                ""type"": ""AdaptiveCard"",
+                ""body"": [
+                    {
+                        ""type"": ""Input.Date"",
+                        ""id"": ""dueDate""
+                    }
+                ],
+                ""actions"": [
+                    {
+                        ""type"": ""Action.Submit"",
+                        ""title"": ""OK""
+                    }
+                ]
+            }
+        },
+        {
+            ""type"": ""Action.ShowCard"",
+            ""title"": ""Comment"",
+            ""card"": {
+                ""type"": ""AdaptiveCard"",
+                ""body"": [
+                    {
+                        ""type"": ""Input.Text"",
+                        ""id"": ""comment"",
+                        ""isMultiline"": true,
+                        ""placeholder"": ""Enter your comment""
+                    }
+                ],
+                ""actions"": [
+                    {
+                        ""type"": ""Action.Submit"",
+                        ""title"": ""OK""
+                    }
+                ]
+            }
+        }
+    ]
+}";
+            List<Task> tasks = new List<Task>();
+            for (var i = 0; i < 10; i++)
+            {
+                var payload = adaptiveCard;
+                var task = Task.Run(() => { RenderCardTask(payload); });
+                tasks.Add(task);
+            }
+
+            try
+            {
+                Task.WaitAll(tasks.ToArray());
+            }
+            catch
+            {
+                // it's perfectly fine card. if there is exception, it is due to concurreny
+                // as it's the only variable.
+                Assert.Fail();
```

Comment:
Nit: Include a message here in case the assertion fails to aid debugging a bit by adding some failure context

Reply:
good catch; it wasn't carried over from the stopgap since they were worked at in parrarrell.


---
In reply to: [341284485](https://github.com/microsoft/AdaptiveCards/pull/3548#discussion_r341284485) [](ancestors = 341284485)